packages/flutter/test/cupertino/route_test.dart

```diff
@@ -1471,6 +1472,29 @@ void main() {
     expect(find.text('first', skipOffstage: false), findsOneWidget);
     expect(find.text('second'), findsOneWidget);
   });
+
+  testWidgets('Popping routes should cancel down events', (WidgetTester tester) async {
+    // The test target, _TestPostRouteCancel, shows on the home route how many
+    // [PointerCancelEvent] it has received. Initially it will show "Home 0".
+    // After the route is popped, it should show "Home 1", meaning it received
+    // one [PointerCancelEvent].
+    await tester.pumpWidget(_TestPostRouteCancel());
+
+    final TestGesture gesture = await tester.createGesture();
+    await gesture.down(tester.getCenter(find.text('Home 0')));
```

Comment:
I think the intentions of the test would be a little clearer if the app would more clearly say what the "0" means. You could for example have an additional text widget that says "PointerCancelEvents: 0". The test would then be self-explanatory and you wouldn't even need the comment above.